src/web/mjs/engine/DiscordPresence.mjs

```diff
@@ -124,20 +126,48 @@ export default class DiscordPresence {
                 this.updateStatus();
             }, 2000);
 
-            // activity can only be set every 15 seconds
+            // activity can only be set every 15 seconds (API limit)
             this.updater = setInterval(() => {
                 this.updateStatus();
             }, 15200);
         });
 
-        this.rpc.login({ clientId: discordPresenceId })
-            .catch (error => {
-                if (error.message.match(/[A-z]+/g).join('').toUpperCase() == 'COULDNOTCONNECT') {
+        try {
+            await this.rpc.login({ clientId: discordPresenceId });
+        } catch (error) {
+            if (typeof error !== 'undefined') { // discord-rpc error handling
+                if (/Could not connect/i.test(error.message)) {
                     console.warn('WARNING: DiscordPresence - Could not connect (Is Discord running?)');
-                } else {
-                    throw new Error(error);
+                    return;
+                }
+
+                if (/RPC_CONNECTION_TIMEOUT/i.test(error.message)) {
+                    console.warn('WARNING: DiscordPresence - RPC connection timed out.');
+                    // Cleanup
+                    this.stopDiscordPresence(true);
+
+                    // Waiting delay for Discord API to allow new connection
+                    setTimeout( () => {
+                        // Re-evaluate if enabled
+                        this._onSettingsChanged();
+                    }, 120000);
+
+                    return;
                 }
-                this.rpc = null;
-            });
+
+                throw new Error(error); // Unknown error
```

Comment:
Since `error` is expected to be of type `Error` it can be re-thrown directly instead of wrapping into another error, e.g. `throw error`